multichain-aggregator/multichain-aggregator-logic/src/services/search.rs

```diff
@@ -121,10 +122,11 @@ pub async fn search_addresses(
             if let Ok(address) = alloy_primitives::Address::from_str(&query) {
                 (vec![address], None)
             } else if domain_name_with_tld_regex().is_match(&query) {
-                let domains = search_domains(
-                    bens_client,
+                let domains = search_domains_cached(
+                    domain_search_cache,
+                    bens_client.clone(),
                     query.clone(),
-                    bens_protocols,
+                    bens_protocols.map(|p| p.to_vec()),
```

Comment:
probably nit, but it might be better to use `Arc` to still be able to pass a "reference" even when owned type is needed for caching
see branch `bragov4ik/review-lok52/multi-remove-contract-search`